website/docs/r/transport_node.html.markdown

```diff
@@ -0,0 +1,276 @@
+      * `default_gateway_address` - (Optional) Default gateway for the node.
+      * `host_id` - (Optional) Host identifier in the specified vcenter server.
+      * `ipv4_assignment_enabled` - (Optional) This flag represents whether IPv4 configuration is enabled or not. Defaults to true.
+      * `management_network_id` - (Required) Portgroup, logical switch identifier or segment path for management network connectivity.
+      * `management_port_subnet` - (Optional) Port subnets for management port. IPv4, IPv6 and Dual Stack Address is supported.
+        * `ip_addresses` - (Required) List of IP addresses.
+        * `prefix_length` - (Required) Subnet Prefix Length.
+      * `reservation_info` - (Optional) Resource reservation settings.
+        * `cpu_reservation_in_mhz` - (Optional) CPU reservation in MHz.
+        * `cpu_reservation_in_shares` - (Optional) CPU reservation in shares. Accepted values - 'EXTRA_HIGH_PRIORITY', 'HIGH_PRIORITY', 'NORMAL_PRIORITY', 'LOW_PRIORITY'. The default value is 'HIGH_PRIORITY'.
+        * `memory_reservation_percentage` - (Optional) Memory reservation percentage.
+      * `storage_id` - (Required) Storage/datastore identifier in the specified vcenter server
+      * `vc_id` - (Required) Vsphere compute identifier for identifying the vcenter server.
+  * `node_settings` - (Required) Current configuration on edge node.
+    * `advanced_configuration` - (Optional) Advanced configuration.
+      * `key` - (Required)
+      * `value` - (Required)
+    * `allow_ssh_root_login` - (Optional) Allow root SSH logins. Defaults to false.
+    * `dns_servers` - (Optional) List of DNS servers.
+    * `enable_ssh` - (Optional) Enable SSH. Defaults to false.
+    * `enable_upt_mode` - (Optional) Enable Uniform Passthrough mode. Defaults to false.
+    * `hostname` - (Required) Host name or FQDN for edge node.
+    * `ntp_servers` - (Optional) List of NTP servers.
+    * `search_domains` - (Optional) List of Search domain names.
+    * `syslog_server` - (Optional) List of Syslog servers.
+      * `log_level` - (Optional) Log level to be redirected. Accepted values - 'EMERGENCY', 'ALERT', 'CRITICAL', 'ERROR', 'WARNING', 'NOTICE', 'INFO' or 'DEBUG'. The default value is 'INFO'.
+      * `name` - (Optional) Display name of the syslog server.
+      * `port` - (Optional) Syslog server port. Defaults to 514.
+      * `protocol` - (Optional) Syslog protocol. Accepted values - 'TCP', 'UDP', 'TLS', 'LI', 'LI_TLS'. The default value is 'UDP'.
+      * `server` - (Required) Server IP or fqdn.
+* `remote_tunnel_endpoint` - (Optional) Configuration for a remote tunnel endpoint.
+  * `host_switch_name` - (Required) The host switch name to be used for the remote tunnel endpoint.
+  * `ip_assignment` - (Required) - Specification for IPs to be used with host switch virtual tunnel endpoints. Should contain exatly one of the below:
+    * `assigned_by_dhcp` - (Optional) Enables DHCP assignment. Should be set to true.
```

Comment:
I like the description :)

Reply:
The API [AssignedByDhcp](https://vdc-download.vmware.com/vmwb-repository/dcr-public/201a09cb-5610-49e6-9d31-7db0920e3a65/65f199d6-8516-4092-9b19-906559f972b5/api_includes/types_AssignedByDhcp.html) object has nothing but resource type - which we do not add in TF. Any suggestion for how to make this more user-understandable? Or replace the bool with anything else?